slatedb/src/utils.rs

```diff
@@ -77,32 +78,28 @@ where
     T: Send + 'static,
     C: FnOnce(&Result<T, SlateDBError>) + Send + 'static,
 {
-    let inner_handle = handle.clone();
-    handle.spawn(async move {
-        let jh = inner_handle.spawn(future);
-        match jh.await {
-            Ok(result) => {
-                cleanup_fn(&result);
-                result
-            }
-            Err(join_err) => {
-                // task panic'd or was cancelled
-                let err = Err(BackgroundTaskPanic(Arc::new(Mutex::new(
-                    join_err
-                        .try_into_panic()
-                        .unwrap_or_else(|_| Box::new("background task was aborted")),
-                ))));
-                cleanup_fn(&err);
-                err
-            }
-        }
-    })
+    // NOTE: It is critical that the future lives as long as the cleanup_fn.
+    //       Otherwise, there is a gap where everything owned by the futured is dropped
```

Comment:
There is a typo here:

```suggestion
    //       Otherwise, there is a gap where everything owned by the future is dropped
```